`pwndbg.gdblib.file.get_file` might trigger an `AssertionError` with a path like `path/to/file`

When passing a path like `path/to/file` to `pwndbg.gdblib.file.get_file`, it might trigger an `AssertionError` here:
https://github.com/pwndbg/pwndbg/blob/93bab21c58557c38f95abec19be06d682074974d/pwndbg/gdblib/file.py#L60-L62
```
pwndbg> info sharedlibrary
From                To                  Syms Read   Shared Object Library
0x00007fb9e847a070  0x00007fb9e84ddc14  Yes (*)     ld-musl-x86_64.so.1
(*): Shared library is missing debugging information.
pwndbg> got -a
Filtering out read-only entries (display them with -r or --show-readonly)

State of the GOT of /work/crewctf/Format muscle/format-muscle:
GOT protection: Full RELRO | Found 0 GOT entries passing the filter

Traceback (most recent call last):
  File "/root/pwndbg/pwndbg/commands/__init__.py", line 200, in __call__
    return self.function(*args, **kwargs)
  File "/root/pwndbg/pwndbg/commands/__init__.py", line 364, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/root/pwndbg/pwndbg/commands/got.py", line 103, in got
    _got(path, accept_readonly, symbol_filter)
  File "/root/pwndbg/pwndbg/commands/got.py", line 116, in _got
    local_path = pwndbg.gdblib.file.get_file(path, try_local_path=True)
  File "/root/pwndbg/pwndbg/gdblib/file.py", line 60, in get_file
    assert path.startswith(("/", "./", "../")) or path.startswith(
AssertionError: get_file called with incorrect path

> /root/pwndbg/pwndbg/gdblib/file.py(60)get_file()
-> assert path.startswith(("/", "./", "../")) or path.startswith(
(Pdb) path
'ld-musl-x86_64.so.1'
```
> This can be reproduced with these binaries: 
[format_muscle_from_CrewCTF.zip](https://github.com/user-attachments/files/16505063/format_muscle_from_CrewCTF.zip)

It seems like this assertion was added a long time ago: https://github.com/pwndbg/pwndbg/commit/790ba574c04cfbf5b12ac8f336cf4fffcdb98faf
It appears to work fine even if we update the assertion to handle `"./"` or `"../"`.
So, maybe we can remove this assertion?
Will there be any side effects if we remove it?

--- a/pwndbg/gdblib/file.py
+++ b/pwndbg/gdblib/file.py
@@ -57,9 +57,11 @@ def get_file(path: str, try_local_path: bool = False) -> str:
     Returns:
         The local path to the file
     """
-    assert path.startswith(("/", "./", "../")) or path.startswith(
-        "target:"
-    ), "get_file called with incorrect path"
+    # Seems like we don't need this check and this check might casue issues, see #2338
+    # Just keep the code commented for now, so we can revert if needed.
+    # assert path.startswith(("/", "./", "../")) or path.startswith(
+    #     "target:"
+    # ), "get_file called with incorrect path"
 
     has_target_prefix = path.startswith("target:")
     if has_target_prefix:

